decompyle3/semantics/transform.py

```diff
@@ -0,0 +1,184 @@
+from decompyle3.show import maybe_show_tree
+from copy import copy
+import sys
+
+from xdis.code import iscode
+from spark_parser import GenericASTTraversal, GenericASTTraversalPruningException
+from decompyle3.scanner import Code
+from decompyle3.parsers.treenode import SyntaxTree
+
+
+class TreeTransform(GenericASTTraversal, object):
+    def __init__(self, scanner, parser, build_ast, show_ast=False):
+        self.showast = show_ast
+        self.build_ast = build_ast
+        self.currentclass = None
+        self.scanner = scanner
+        self.p = parser
+        self.hide_internal = False
```

Comment:
I made a mistake here - the default should be `True` (we want stuff like `__doc__ = "this is a docstring" hidden).

Reply:
It seems setting `hide_internal` to `True` fixed the `MAKE_FUNCTION` parse error. Thanks!

Reply:
but we have an issue in handling args, for following code:

```python
class BoundedSemaphore(Semaphore):
    def __init__(self, value=1, *, loop=None):
        super().__init__(value, loop=loop)
```

we get:
```python
class BoundedSemaphore(Semaphore):

    def __init__(self, value=1, loop=None):
        super().__init__(value, loop=loop)
```

I will fix this issue.

Reply:
Thanks! Also a new issue is popping up in handling f_string:

```
def _repr_fn(fields):
    return testit'__repr__'('self', )[
#               ^^  missing surrounding ()'s  
     'return xx + f"(' + ', '.join([f"{f}={{self.{f}!r}}" for f in fields]) + ')"']
```

With respect to the scanner, let's make those changes in the `assert2` branch since that's where this probably belongs and we can get that in sooner. Thanks again. 

This PR is probably going to take a while because it is a bit disruptive and needs work in a number of areas - see checklist.